src/Features/VisualBasic/Portable/Completion/CompletionProviders/PartialTypeCompletionProvider.vb

```diff
@@ -57,29 +63,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
             End If
         End Function
 
-        Private Async Function CreateItemsAsync(document As Document, position As Integer, span As TextSpan, token As SyntaxToken, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of CompletionItem))
+        Private Async Function CreateItemsAsync(document As Document, position As Integer, span As TextSpan, token As SyntaxToken, typeStatement As TypeStatementSyntax, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of CompletionItem))
             Dim semanticModel = Await document.GetSemanticModelForNodeAsync(token.Parent, cancellationToken).ConfigureAwait(False)
 
-            ' Unless the enclosing symbol is already the global namespace, we want to get it's enclosing symbol
-            ' in order to suggest partial types in our namespace.
-            Dim enclosingSymbol = semanticModel.GetEnclosingSymbol(position, cancellationToken)
-            Dim enclosingNamespace = TryCast(enclosingSymbol, INamespaceSymbol)
-            If Not (enclosingNamespace IsNot Nothing AndAlso enclosingNamespace.IsGlobalNamespace) Then
-                enclosingSymbol = enclosingSymbol.ContainingSymbol
-            End If
-
-            Dim displayService = document.GetLanguageService(Of ISymbolDisplayService)()
+            Dim declaredSymbol = semanticModel.GetDeclaredSymbol(typeStatement, cancellationToken)
```

Comment:
It seemed less arbitrary to base the lookup on the syntax node, which we already got when checking for the `Partial` modifier. Of course, that didn't initially work out either. Refer to my change to the semantic model.